### DISCUSSION TITLE
redirect url with #! symbols

### DISCUSSION THREAD

**Discussion by ricardo-fnd** [vercel/next.js#38731] [Help]
Hi guys!
I'm refactoring an application developed in Angular and I need to keep the same urls. One example of an url on the old app is `http://localhost:8080/login#!/sso`
When this url is visited I want to redirect to the new url, this time without the `#!` symbols, like, `https://localhost:8080/login/sso`.

I've tried to do redirects and rewrites, but it seems that the symbols are not read by next. Any ideias here?
thanks.

**Comment by GabenGar** [comment]
Do redirects on client.

**Reply by ricardo-fnd** [reply]
yeah, I was trying to avoid that as I wanted to do a redirect in the server, but I guess it is the only choice.
